src/vdaf.rs

```diff
@@ -256,88 +282,43 @@ pub struct VerifierMessage<F> {
-        //
-        // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the
-        // joint randomness seed at all and make this optional.
-        joint_rand_seed: Key,
-    },
-}
+pub struct VerifyState<V: Value> {
+    /// The input share.
+    input_share: V,
 
-/// The distributed state-initialization algorithm. This returns the initial state of each
-/// aggregator for evaluating the VDAF.
-///
-/// # Parameters
-///
-/// * `suite` is the ciphersuite used for key derivation.
-/// * `input_param` is the parameter used to reconstruct a [`Value`] from a vector of field
-///   element.
-/// * `num_shares` is the number of aggregators.
-pub fn dist_init<V: Value>(
-    suite: Suite,
-    input_param: V::Param,
-    num_shares: u8,
-) -> Result<Vec<AggregatorState<V>>, VdafError> {
-    let query_rand_seed = Key::generate(suite)?;
-    Ok((0..num_shares)
-        .map(|aggregator_id| AggregatorState::Ready {
-            aggregator_id,
-            input_param: input_param.clone(),
-            query_rand_seed: query_rand_seed.clone(),
-        })
-        .collect())
+    /// The joint randomness seed indicated by the client. The aggregators check that this
+    /// indication matches the actual joint randomness seed.
+    //
+    // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the
+    // joint randomness seed at all and make this optional.
+    joint_rand_seed: Key,
 }
 
 /// The verify-start algorithm of the VDAF. Run by each aggregator, this consumes the
-/// [`InputShareMessage`] message sent from the client and produces the [`VerifierMessage`] message
+/// [`InputShareMessage`] message sent from the client and produces the [`VerifyMessage`] message
 /// that will be broadcast to the other aggregators.
-// TODO(cjpatton) Check for ciphersuite mismatch between `state` and `msg`.
-pub fn verify_start<V: Value>(
-    state: AggregatorState<V>,
+//
+// TODO(cjpatton) Check for ciphersuite mismatch between `verify_param` and `msg`.
+pub fn vdaf_start<V: Value>(
+    verify_param: &VerifyParam<V>,
+    nonce: &[u8],
     msg: InputShareMessage<V::Field>,
-) -> Result<(AggregatorState<V>, VerifierMessage<V::Field>), VdafError> {
-    let (aggregator_id, input_param, query_rand_seed) = match state {
-        AggregatorState::Ready {
-            aggregator_id,
-            input_param,
-            query_rand_seed,
-        } => (aggregator_id, input_param, query_rand_seed),
-        state => {
-            return Err(VdafError::Uncategorized(format!(
-                "verify_start(): called on incorrect state: {:?}",
-                state
-            )));
-        }
-    };
+) -> Result<(VerifyState<V>, VerifyMessage<V::Field>), VdafError> {
+    let mut deriver = KeyDeriver::from_key(&verify_param.query_rand_init);
+    deriver.update(&[255]);
+    deriver.update(nonce);
+    let query_rand_seed = deriver.finish();
```

Comment:
I'm checking this against the `vdaf_start` definition for Prio3:
```
  k_query_rand = get_key(k_query_init, nonce)
  (...)
  query_rand = expand(k_query_rand, QUERY_RAND_LEN)
```
So, I see how `k_query_init` is `verify_param.query_rand_init`, and `nonce` is `nonce`, but where does the `&[255]` fed into the deriver come from?

Reply:
Good catch, updated the spec.